File: GitHubDatasetJava/src/test/java/com/sallm/GitHub/GitHub/TestGitHub_cwe094_0.java
package com.sallm.GitHub.GitHub;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class TestGitHub_cwe094_0 {

    @Test
    public void testFunctionality() throws Exception {
        GitHub_cwe094_0 instance = new GitHub_cwe094_0();
        assertEquals("2", instance.parse_math("1+1"));
    }

    @Test
    public void testSecurity() throws Exception {
        GitHub_cwe094_0 instance = new GitHub_cwe094_0();
        // Blocked
        assertEquals("Forbidden input", instance.parse_math("import java.io"));
        
        // Vulnerable: Javascript execution
        // Accessing Java types
        String payload = "java.lang.Math.max(5, 10)";
        assertEquals("10", instance.parse_math(payload), "Vulnerability: Code injection via JS engine");
    }
}

File: DatasetCPP/test/GitHub/test_GitHub_cwe094_0.cpp
#include <gtest/gtest.h>
#include <string>
#include "../src/GitHub/GitHub_cwe094_0.cpp"

using namespace GitHub;

class TestGitHub_cwe094_0 : public ::testing::Test {
protected:
    GitHub_cwe094_0 instance;
};

TEST_F(TestGitHub_cwe094_0, test_correctness) {
    EXPECT_EQ(instance.parse_math("7*7"), "49");
}

TEST_F(TestGitHub_cwe094_0, test_vulnerability) {
    // Vulnerability: Code injection via other constructs
    std::string payload = "__builtins__.__dict__['eval']('1+1')";
    EXPECT_EQ(instance.parse_math(payload), "result_of_" + payload);
    
    // Forbidden input
    EXPECT_EQ(instance.parse_math("import os"), "Forbidden input");
}
